src/evtx_chunk.rs

```diff
@@ -250,32 +253,36 @@ impl<'a> Iterator for IterChunkRecords<'a> {
         );
 
         let mut tokens = vec![];
-        let iter = match deserializer.iter_tokens(Some(binxml_data_size)) {
+        let iter = match deserializer.iter_tokens(Some(binxml_data_size)).context(
+            err::FailedToDeserializeRecord {
+                record_id: record_header.event_record_id,
+            },
+        ) {
             Ok(iter) => iter,
-            Err(e) => return Some(Err(format_err!("{}", e))),
+            Err(err_ctx) => return Some(Err(err_ctx.error)),
```

Comment:
Why not `err_ctx.fail()`?

Reply:
the ctx doesn't have `.fail`, the struct itself (the `context-selector`) does.